In [ ]:
# Importing libraries

import pandas as pd 
import numpy as np 
import sys 
from pathlib import Path
from sklearn.model_selection import train_test_split

In [ ]:
# Using original data path 
sys.path.append(str(Path.cwd().parents[0]))

In [ ]:
# Uploading libraries/functions built from previous files 

from src.dataset import finalizing_dataset
from src.features import drop_col_for_train
from src.preprocessing import column_type, building_pipeline

In [ ]:
# using the finalized dataset function to import data 

df = finalizing_dataset()

In [5]:
df.head()

encounter_id  patient_nbr             race  gender      age  \
0       2278392      8222157        Caucasian  Female   [0-10)   
1        149190     55629189        Caucasian  Female  [10-20)   
2         64410     86047875  AfricanAmerican  Female  [20-30)   
3        500364     82442376        Caucasian    Male  [30-40)   
4         16680     42519267        Caucasian    Male  [40-50)   

   admission_type_id  discharge_disposition_id  admission_source_id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   

   time_in_hospital  num_lab_procedures  ...  diabetesMed  readmitted  \
0                 1                  41  ...           No          NO   
1                 3                  59  ...          Yes         >30   
2                 2                  11  ...          Yes          NO   
3                 2                  44  ...          Yes          NO   
4                 1                  51  ...          Yes          NO   

   readmitted_30  num_inpatient  num_emergency num_outpatient total_visits  \
0              0              0              0              0            0   
1              0              0              0              0            0   
2              0              1              0              2            3   
3              0              0              0              0            0   
4              0              0              0              0            0   

  num_medication_change  insulin_used has_diabetes_complications  
0                     0             0                          1  
1                     1             1                          0  
2                     0             0                          0  
3                     1             1                          1  
4                     0             1                          0  

[5 rows x 55 columns]

In [6]:
df.shape

(101766, 55)

In [ ]:
# Setting the target value and splitting variables into X and y for further ml 
target = 'readmitted_30'

y = df[target]
X = df.drop(columns=target)

In [ ]:
# Getting the column type of each of our columns 

categorical_cols, numerical_cols = column_type(X)

display(len(categorical_cols))
display(len(numerical_cols))

34

20

Binary values were treated as categorical variables. Continuous values treated as numerical variables. 

In [ ]:
# Splitting the data into test and train for ml inputs 

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size =0.2, stratify=y, random_state = 42
)

In [ ]:
# displaying the means of the y train and test 

display(y_train.mean())
display(y_test.mean())

0.11160516877118852

0.11157512036946055

Target variable is imbalanced. Test/Train split was used to preserve class distribution. 
Train rate: 11.16%
Test rate: 11.15%

In [ ]:
# Using the building pipeline function on our categorical and numerical columns 

preprocessor = building_pipeline(
    categorical_cols, 
    numerical_cols
)

preprocessor

ColumnTransformer(transformers=[('num',
                                 Pipeline(steps=[('scaler', StandardScaler())]),
                                 ['encounter_id', 'patient_nbr',
                                  'admission_type_id',
                                  'discharge_disposition_id',
                                  'admission_source_id', 'time_in_hospital',
                                  'num_lab_procedures', 'num_procedures',
                                  'num_medications', 'number_outpatient',
                                  'number_emergency', 'number_inpatient',
                                  'number_diagnoses', 'num_inpatient...
                                  'diag_3', 'max_glu_serum', 'A1Cresult',
                                  'metformin', 'repaglinide', 'nateglinide',
                                  'chlorpropamide', 'glimepiride',
                                  'acetohexamide', 'glipizide', 'glyburide',
                                  'tolbutamide', 'pioglitazone',
                                  'rosiglitazone', 'acarbose', 'miglitol',
                                  'troglitazone', 'tolazamide', 'examide',
                                  'citoglipton', 'insulin',
                                  'glyburide-metformin', 'glipizide-metformin',
                                  'glimepiride-pioglitazone',
                                  'metformin-rosiglitazone', ...])])

In [ ]:
# Showing the test and train split of our X value 

display(X_train.shape)
display(X_test.shape)

(81412, 54)

(20354, 54)